### ISSUE TITLE
Plugin support for Windows editing via Docker? 

### ISSUE THREAD

**Issue by andmos** [sydro/atom-ansible-vault#24]
First off, I love this plugin - so thanks so much for the project! 

I mainly use OS X when I administrate Ansible stuff, but I also use a Windows 10 desktop. 
I was doodling on a possible solution for this plugin accessing Ansible Vault from Windows via Docker. 
If a Linux container goes in the background and exposes the Vault binary and the encrypted files, my unqualified guess is that it should be able to get it working.

**Comment by sydro** [comment]
Hi, docker isn't a best solution (maybe one), but i'm trying some tests to add support to windows anyway

**Comment by andmos** [comment]
Yeah after some thinking I figured that out myself. As long as ansible-vault has to be run in a *NIX shell we have to be creative. How about running it via [Linux subsystem for Windows](https://msdn.microsoft.com/en-us/commandline/wsl/install_guide) and simply spawning the command in the Ubuntu shell if it is detected to be running on Windows? A person running Ansible on Windows in some way has to be creative in the first place, so he or she would be able to understand that. Personally I run Ansible via Vagrant on my Windows machine, but my colleagues have it installed via the Linux Subsystem.

**Comment by sydro** [comment]
Hi @andmos,
I made some tests and finally it works.
I push a new branch [windows-support](https://github.com/sydro/atom-ansible-vault/tree/windows-support) with a patch and a [README-Windows.md](https://github.com/sydro/atom-ansible-vault/blob/windows-support/README-Windows.md) file that contains guidelines.

If you want test it i will be pleased :+1: 

Probably i will test docker solution too!

**Comment by andmos** [comment]
Brilliant! Nice work! I will give it a try on monday!

**Comment by sydro** [comment]
Any news? Did you try it?

**Comment by cvdabbeele** [comment]
Hi sydro
I'd be happy to give this great feature a try on my windows 10 workstation.
Can you provide the ansible-vault.bat file ? 
It seems to be missing at the link referenced at: https://github.com/sydro/atom-ansible-vault/blob/windows-support/README-Windows.md 
thanks so much for this very valuable feature 
chris

**Comment by sydro** [comment]
Sorry, the link to ansible-vault.bat was broken. I fixed it

**Comment by cvdabbeele** [comment]
Thank you so much Kyle
I'm seeing mixed results, but I may have a bit of an edge case...
File Open with auto-decrypt enabled (with pw in /home/ubuntu/pass.txt on WSL) works great.  Cool !
But "Alt-File-Save" saves the file as clear text (should I maybe use another command?)
Using Windows10 with Windows subsystem for Linux (https://docs.microsoft.com/en-us/windows/wsl/install-win10 )   
The secrets file is on my Windows filesystem (D:\...) and Atom saves it locally and then syncs it to a remote Ubuntu system. (what could possible go wrong :-)  )  Both save instances become clear-text
I'm still discovering atom, maybe I do something wrong.  
Also the key-binding ctrl-alt-0 is enabled but that doesn't seem to do anything either...

**Comment by sydro** [comment]
I need to rebase this branch on master, there'are some changes that fix many bugs and this branch is very old. 
i will do it and push all after test. 

Thanks for your tests :+1:

**Comment by andmos** [comment]
I'm so sorry I didn't catch this, since I started the whole thing... But great if you guys got it working, I'll have a look myself!
